qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -604,31 +604,32 @@ def metadata_headers(cls):
             qdb.sql_connection.TRN.add(sql)
             return qdb.sql_connection.TRN.execute_fetchflatten()
 
-    def _common_delete_sample_steps(self, sample_name):
+    def _common_delete_sample_steps(self, sample_names):
         r"""Executes the common delete sample steps
 
         Parameters
         ----------
-        sample_name : str
-            The sample name to be erased
+        sample_names : list of str
+            The sample names to be erased
 
         Raises
         ------
         QiitaDBUnknownIDError
-            If the `sample_name` doesn't exist
+            If the `sample_names` doesn't exist
         """
-        if sample_name not in self.keys():
-            raise qdb.exceptions.QiitaDBUnknownIDError(sample_name, self._id)
+        keys = self.keys()
+        missing = [sn for sn in sample_names if sn not in keys]
+        if missing:
+            raise qdb.exceptions.QiitaDBUnknownIDError(missing, self._id)
```

Comment:
Looks like this exception class is not meant to accept a list of strings. Maybe using a different subclass or formatting the missing sample names in a different way. Though, even then the text doesn't make a lot of sense, see [this line here](https://github.com/biocore/qiita/blob/master/qiita_db/exceptions.py#L81).